server/network_server.c

```diff
@@ -1121,6 +1126,19 @@ _cb_worker(struct network_client_s *clt, struct network_server_s *srv)
 		return;
 	}
 
+	/* The event stayed *really* long in the queue of the thread pool.
+	 * Let's close the connection, and let the client retry it's request. */
+	if (clt->events & CLT_READ) {
+		const gint64 now = oio_ext_monotonic_time();
+		if (clt->time.evt_in < OLDEST(now, G_TIME_SPAN_SECOND)) {
+			GRID_INFO("SLOW fd %d peer %s delay %"G_GINT64_FORMAT"ms",
+					clt->fd, clt->peer_name,
+					(now - clt->time.evt_in) / 1000);
```

Comment:
I agree that warnings may fill the logs. Maybe we could warn from another thread, at most once every N seconds, if the `gq_count_overloaded` counter increases.